src/XenCert/StorageHandlerUtil.py

```diff
@@ -328,7 +307,7 @@ def get_hba_information(session, storage_conf, sr_type="lvmohba"):
 		    else:
 			    list.append(map)
 		
-		bd_list = dom.getElementsByTagName("BlockDevice")
+		bd_list = getlist(xmlstr, "BlockDevice")
```

Comment:
Here makes xmlstr is parsed twice. 
- You can let getlist() return only the 'dom' object.
- Also please rename getlist(), e.g. parse_probe_xml_data()

Finally like
`
def parse_probe_xml_data(xmlstr):
    xmlstr = xmlstr.lstrip()
    xmlstr = xmlstr.lstrip('\'')
    xmlstr = xmlstr.rstrip()
    xmlstr = xmlstr.rstrip('\]')
    xmlstr = xmlstr.rstrip('\'')
    xmlstr = xmlstr.replace('\\n', '')
    xmlstr = xmlstr.replace('\\t', '')
    xencert_print("Got the probe xml as: %s" % xmlstr)
    return xml.dom.minidom.parseString(xmlstr)
`
then change all callers

Reply:
ok, that's reasonable.